internal/oc/extract.go

```diff
@@ -0,0 +1,50 @@
+package oc
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/openshift/assisted-service/pkg/executer"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=extract.go -package=oc -destination=mock_extract.go
+type Extracter interface {
+	Extract(log logrus.FieldLogger, imageIndexPath string, openshiftVersion string, filePath string, pullSecret string) (string, error)
+	ExtractDatabaseIndex(log logrus.FieldLogger, openshiftVersion string, pullSecret string) (string, error)
+}
+
+type extract struct {
+	executer executer.Executer
+	config   Config
+}
+
+func NewExtracter(executer executer.Executer, config Config) Extracter {
+	return &extract{executer, config}
+}
+
+const (
+	imageIndex           = "registry.redhat.io/redhat/redhat-operator-index"
+	dbFile               = "/database/index.db"
+	templateImageExtract = "oc image extract %s:v%s --path=%s:%s"
+)
+
+// ExtractDatabaseIndex extracts the databse idnex from the redhat-operator-index
+func (r *extract) ExtractDatabaseIndex(log logrus.FieldLogger, openshiftVersion string, pullSecret string) (string, error) {
```

Comment:
Who calls this function?

Reply:
Same as above, this will be called in validation method. The output of this method is later used to initialize the sqllite struct to query the operator versions.